File: maximum.js
// Function to find the maximum value in an array
function findMax(arr) {
    // Handle edge case: empty array
    if (arr.length === 0) {
        throw new Error("Array is empty. Cannot find maximum value.");
    }

    // Initialize the maximum value with the first element of the array
    let max = arr[0];

    // Loop through the array to find the maximum
    for (let i = 1; i < arr.length; i++) {
        if (arr[i] > max) {
            max = arr[i]; // Update max if the current element is greater
        }
    }

    return max; // Return the maximum value
}

// Example usage:
console.log(findMax([3, 1, 4, 1, 5, 9])); // Output: 9

File: maximum.py
# Function to find the maximum value in an array
def find_max(arr):
    # Handle edge case: empty array
    if len(arr) == 0:
        raise ValueError("Array is empty. Cannot find maximum value.")

    # Initialize the maximum value with the first element of the array
    max_val = arr[0]

    # Loop through the array to find the maximum
    for num in arr[1:]:
        if num > max_val:
            max_val = num  # Update max_val if the current element is greater

    return max_val  # Return the maximum value

# Example usage:
print(find_max([3, 1, 4, 1, 5, 9]))  # Output: 9
